File: Mnema.Models/Entities/Content/Subscription.cs
using System;
using Mnema.API.Repositories;
using Mnema.Common;
using Mnema.Models.Entities.Interfaces;
using Mnema.Models.Enums;

namespace Mnema.Models.Entities.Content;

[Obsolete("Use MonitoredSeries")]
public class Subscription: IEntityDate, IDatabaseEntity
{
    public Guid Id { get; set; }

    /// <summary>
    ///     The external content id
    /// </summary>
    public required string ContentId { get; set; }

    /// <summary>
    ///     Title given by the user, defaults to the series name
    /// </summary>
    public required string Title { get; set; }

    /// <summary>
    ///     The directory to download the content in
    /// </summary>
    public required string BaseDir { get; set; }

    public required Provider Provider { get; set; }

    [JsonColumn]
    public required MetadataBag Metadata { get; set; }

    public required SubscriptionStatus Status { get; set; }

    public DateTime CreatedUtc { get; set; }
    public DateTime LastModifiedUtc { get; set; }
}

public enum SubscriptionStatus
{
    Enabled = 0,
    Disabled = 1,
}

File: Mnema.Frontend/Web/src/app/_models/subscription.ts
import {Provider} from "./page";
import {MetadataBag} from "./search";

export type Subscription = {
  id: string;
  contentId: string;
  refreshFrequency: RefreshFrequency;
  title: string;
  baseDir: string;
  lastRun: Date;
  lastRunSuccess: boolean;
  nextRun: Date;
  provider: Provider;
  metadata: MetadataBag;
}

export enum RefreshFrequency {
  Day = 2,
  Week,
  Month,
}
